tests/tests_catapult.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+ROOT_DIR="$(git rev-parse --show-toplevel)"
+
+[ ! -d "shunit2" ] && git clone https://github.com/kward/shunit2.git
+
+setUp() {
+    export CLUSTER_NAME=test
+    export ROOT_DIR="$(git rev-parse --show-toplevel)"
+    pushd "$ROOT_DIR"
+}
+
+tearDown() {
+    export ROOT_DIR="$(git rev-parse --show-toplevel)"
+    pushd "$ROOT_DIR"
+    rm -rf buildtest
+}
+
+# Tests creation and deletion of build directory
+testBuilddir() {
+  rm -rf buildtest
+  make buildir
+  assertTrue 'create buildir' "[ -d 'buildtest' ]"
+  ENVRC="$(cat "$PWD"/buildtest/.envrc)"
+  assertContains 'contains KUBECONFIG' "$ENVRC" 'KUBECONFIG="$(pwd)"/kubeconfig'
+  assertContains 'contains CLUSTER_NAME' "$ENVRC" 'CLUSTER_NAME=test'
+  assertContains 'contains CF_HOME' "$ENVRC" 'CF_HOME="$(pwd)"'
+  make clean
+  assertTrue 'clean buildir' "[ ! -d 'buildtest' ]"
+}
+
+# Tests values generation
+testConfig() {
+  rm -rf buildtest
+  make buildir
+  DEFAULT_STACK=sle CLUSTER_PASSWORD=test123 make -C scripts/scf gen-config
```

Comment:
Why no gen-config on the top-level make target ? Imho even if it should be treated as internal we still should have a bridge(or whatever) that allows us to call such targets with more ease. Even a simple "agreement" like: all internal targets should be in form `private-(module)-(action)`, but that's just my 2c

Reply:
That could be definitely done, haven't made more internal api separation than the makefiles inside of the folders, which I think is useful for newcomers doing just `make ^TAB`.
I'm concerned about adding more things to this PR, I would like to merge it asao to reduce the bottleneck/rebases

Reply:
I know.. and do not want to push this much over, I will try to add few commits on top of this work and give it a try on the CI